reports/ccip-functions-report.md

```diff
@@ -2654,3 +2655,38 @@ Invoking `SSTORE`operations in loops may lead to Out-of-gas errors. Use a local
 
 
 
+## L-16: Unused Imports
+
+Redundant import statement. Consider removing it.
+
+<details><summary>4 Found Instances</summary>
+
+
+- Found in src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol [Line: 5](../tests/ccip-contracts/contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol#L5)
+
+	```solidity
+	import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
+	```
+
+- Found in src/v0.8/functions/v1_0_0/FunctionsCoordinator.sol [Line: 5](../tests/ccip-contracts/contracts/src/v0.8/functions/v1_0_0/FunctionsCoordinator.sol#L5)
+
+	```solidity
+	import {ITypeAndVersion} from "../../shared/interfaces/ITypeAndVersion.sol";
+	```
+
+- Found in src/v0.8/functions/v1_1_0/FunctionsCoordinator.sol [Line: 5](../tests/ccip-contracts/contracts/src/v0.8/functions/v1_1_0/FunctionsCoordinator.sol#L5)
+
+	```solidity
+	import {ITypeAndVersion} from "../../shared/interfaces/ITypeAndVersion.sol";
+	```
+
+- Found in src/v0.8/functions/v1_3_0/FunctionsCoordinator.sol [Line: 5](../tests/ccip-contracts/contracts/src/v0.8/functions/v1_3_0/FunctionsCoordinator.sol#L5)
+
+	```solidity
+	import {ITypeAndVersion} from "../../shared/interfaces/ITypeAndVersion.sol";
+	```
```

Comment:
These are used in documentation comments. Should we capture them or not ?

Reply:
Ah this is a false positive @TilakMaddy. That import _is_ actually used on this line:

```solidity
  /// @inheritdoc ITypeAndVersion
  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
  string public constant override typeAndVersion = "Functions Coordinator v1.3.0";
  ```
  
  The `typeAndVersion` variable overrides and implements a function defined in ITypeAndVersion.

Reply:
CORRECTION: This is not a false positive like first thought!!
`ITypeAndVersion` is actually inherited through multiple levels via `OCR2Base` -> `OCR2Abstract`.

It looks as though it is specifically imported to inherit the documentation block.

Let's capture for now. If this actually is a hassle in future, we can also check structured documentation blocks for these symbols.